src/targets.rs

```diff
@@ -697,6 +702,36 @@ impl FromStr for OperatingSystem {
     type Err = ();
 
     fn from_str(s: &str) -> Result<Self, ()> {
+        if s.starts_with("macosx") {
+            // Parse operating system names like `macosx10.7.0`.
+            let s = &s["macosx".len()..];
+            let mut parts = s.split(".").map(|num| num.parse::<u16>());
```

Comment:
Does `u16`'s `FromStr` accept hexadecimal syntax like `0x...`? If so, we should do extra validation so that only accept the syntax used in other triple-using tools.

Reply:
No, it has to be base 10.